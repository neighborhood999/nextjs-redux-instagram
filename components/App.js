import React from 'react';
import Head from 'next/head';
import { Container, Menu, Icon } from 'semantic-ui-react';

const App = ({ children }) =>
  <div id="app">
    <Head>
      <title>Instagram - Building by Nextjs & redux-saga</title>
      <meta name="viewport" content="initial-scale=1.0, width=device-width" />
      <link
        rel="stylesheet"
        href="//cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.2.2/semantic.min.css"
      />
    </Head>
    <Menu size="massive">
      <Menu.Item>
        <Icon name="instagram" circular />
      </Menu.Item>
    </Menu>
    <Container>
      {children}
    </Container>
  </div>;

export default App;
