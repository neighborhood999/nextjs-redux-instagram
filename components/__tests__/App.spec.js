import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import App from '../App';

describe('App Component', () => {
  test('should render children component', () => {
    const wrapper = shallow(
      <App>
        <div>Hello World</div>
      </App>
    );
    expect(wrapper.contains(<div>Hello World</div>)).toBeTruthy();

    const component = renderer.create(
      <App>
        <div>Hello World</div>
      </App>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
