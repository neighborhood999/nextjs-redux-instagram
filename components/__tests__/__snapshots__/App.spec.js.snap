// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render children component 1`] = `
<div
  id="app"
>
  <div
    className="ui massive menu"
  >
    <div
      className="item"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="instagram circular icon"
      />
    </div>
  </div>
  <div
    className="ui container"
  >
    <div>
      Hello World
    </div>
  </div>
</div>
`;
